src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -0,0 +1,155 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
+import SuggestionsListItem, { SuggestionItem } from './SuggestionsListItem';
+import Spinner from '../Spinner';
+import Text, { TextSize, TextColor } from '../Text';
+import Callout from '../Callout';
+import Block, { GutterSize } from '../Block';
+import './SuggestionsList.css';
+
+export interface SuggestionsListProps extends NestableBaseComponentProps {
+  /**
+   * Flag to indicate that a search is in flight
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Text to show next to the loading spinner.
+   */
+  loadingText: string;
+  /**
+   * Text to show when the results are empty for the active search.
+   */
+  noResultsText: string;
+  /**
+   * The groups of items to render.
+   */
+  groupedItems?: SuggestionItemGroupProps[];
+  /**
+   * The index of current keyboard selection.
+   */
+  selectedId?: string | number;
+  /**
+   * The active search that produced the current state.
+   */
+  searchText: string;
+  /**
+   * Called when an item in a group is selected by clicking.
+   */
+  onItemSelected(item: SuggestionItem): void;
+  /**
+   * The target that the Callout should try to position itself based on.
+   */
+  target: HTMLElement;
```

Comment:
I'd sort these attributes a bit better. Mandatory attributes at the top, then optional attributes, then even handlers (`onItemSelected`).